### ISSUE TITLE
Data sources for EBS default encryption

### ISSUE THREAD

**Issue by ewbankkit** [terraform-providers/terraform-provider-aws#8772]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Add new data sources for [EBS default encryption](https://aws.amazon.com/about-aws/whats-new/2019/05/with-a-single-setting-you-can-encrypt-all-new-amazon-ebs-volumes/).

### New or Affected Resource(s)

* `aws_ebs_encryption_by_default` data source
* `aws_ebs_default_kms_key` data source

### Potential Terraform Configuration

```hcl
data "aws_ebs_encryption_by_default" "ebs" {}

data "aws_ebs_default_kms_key" "ebs" {}
```

### References

* https://github.com/terraform-providers/terraform-provider-aws/issues/8760

**Comment by Jukie** [comment]
I'd like to work on this if someone else isn't already

**Comment by Jukie** [comment]
What value would a data source for `aws_ebs_encryption_by_default` add and what would the potential parameters be?

**Comment by Jukie** [comment]
This will be my first time adding a new item rather than modifying an existing one. Is there anything special in order to successfully build & test my changes?
```
jukie@Jukies-MacBook-Pro terraform-provider-aws (issue-8772) $ pwd
/Users/jukie/Documents/github/terraform-provider-aws
jukie@Jukies-MacBook-Pro terraform-provider-aws (issue-8772) $ make build
==> Checking that code complies with gofmt requirements...
go install
jukie@Jukies-MacBook-Pro terraform-provider-aws (issue-8772) $ make testacc TEST=./aws/ TESTARGS='-run=TestAccDataSourceAwsEBSEncryptionByDefault_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -parallel 20 -run=TestAccDataSourceAwsEBSEncryptionByDefault_basic -timeout 120m
=== RUN   TestAccDataSourceAwsEBSEncryptionByDefault_basic
=== PAUSE TestAccDataSourceAwsEBSEncryptionByDefault_basic
=== CONT  TestAccDataSourceAwsEBSEncryptionByDefault_basic
--- FAIL: TestAccDataSourceAwsEBSEncryptionByDefault_basic (1.15s)
    testing.go:568: Step 0 error: config is invalid: Invalid data source: The provider provider.aws does not support data source "aws_ebs_encryption_by_default".
FAIL
FAIL	github.com/terraform-providers/terraform-provider-aws/aws	1.205s
make: *** [testacc] Error 1
```

I wouldn't be surprised if I just have a typo somewhere as well...
https://github.com/Jukie/terraform-provider-aws/tree/issue-8772

**Comment by bflad** [comment]
Support for the below two new data sources has been merged and will release with version 2.17.0 of the Terraform AWS Provider, likely later this week. 👍 

* **New Data Source:** `aws_ebs_default_kms_key`
* **New Data Source:** `aws_ebs_encryption_by_default`

**Comment by bflad** [comment]
This has been released in [version 2.17.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#2170-june-28-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

For further feature requests or bug reports with this functionality, please create a [new GitHub issue](https://github.com/terraform-providers/terraform-provider-aws/issues/new/choose) following the template for triage. Thanks!
